Dev

AnnaHost opened the pull request.

Commit:
- добавлен класс для курьера
- скопирован класс для генерации уникального пользователя
 - написаны тесты для функционала создания курьера
 - частично написаны тесты для авторизации курьера

Commit:
- добавлен класс для тестов оформления заказа
- написаны тесты для проверки оформления заказа согласно требованиям
- добавлен класс для тестов списка заказов

Commit:
- добавлены методы для удаления созданных объектов
- добавлены проверки во все тесты

Commit:
- добавлена папка с отчётами allure

egorvas commented on src/main/java/Order.java:
```diff
@@ -0,0 +1,117 @@
+        this.color = color;
+    }
+
+    public Order(String stringColor) {
```
Вместо того чтобы самому придумывать тестовые строки я рекомендую использовать модули для генерации данных, например java faker: https://github.com/DiUS/java-faker
С его помощью ты убрать большую часть хардкора из твоих тестов.

egorvas commented on src/main/java/Courier.java:
```diff
@@ -0,0 +1,73 @@
+        this.firstName = firstName;
+    }
+
+    public void deleteCourier(Courier courier) {
```
Я рекомендую запросы отделять от моделей и выносить их в степы (делать аннотацию @Step).  В этом случае твой отчет будет выглядеть нагляднее.

egorvas commented on src/test/java/CourierLoginTest.java:
```diff
@@ -0,0 +1,137 @@
+    public void courierCanLogin() {
+        ScooterRegisterCourier courier = new ScooterRegisterCourier();
+        ArrayList<String> list = courier.registerNewCourierAndReturnLoginPassword();
+        String loginRequestBody = "{\"login\":\"" + list.get(0) + "\","
```
Я ярый противник такого формирования тела запроса, все эти скобочки, кавычки и слеши это перебор. В реальной жизни так никто не делает. Ты же подключила gson вот его и используй лучше.

egorvas commented on src/test/java/CreateOrderTest.java:
```diff
@@ -0,0 +1,66 @@
+
+    @Before
+    public void setUp() {
+        RestAssured.baseURI = "https://qa-scooter.praktikum-services.ru";
```
baseURI лучше выносить в одно общее место для всех тестов. На практике твои тесты будут запускаться на разных энвах: дев, прод, тест и тп и везде будут разные url. По этому чтобы в каждом тесте не менять baseURI лучше это делать один раз для всех тестов.

AnnaHost replied:
Где я могу почитать об этом? В голову ничего не приходит и в интернете информацию не нашла. Думаю, что можно как-то в pom указать это.

Commit:
- Добавлены @Step во все тесты
- Добавлена автоматическая генерация входных данных для заказа с помощью Faker
- Убран хардкод для генерации тела запроса в CourierLoginTest. Используется класс Courier

Commit:
- Полностью переделана структура объектов
- Обновлена структура тестов

Commit:
- Полностью переделана структура объектов
- Обновлена структура тестов
